packages/html/src/events/mouse.rs

```diff
@@ -150,131 +106,91 @@ impl_event! {
+        self.inner.page_coordinates()
     }
 
     /// The event's coordinates relative to the entire screen. This takes into account the window's offset.
-    pub fn screen_coordinates(&self) -> ScreenPoint {
-        #[allow(deprecated)]
-        ScreenPoint::new(self.screen_x.into(), self.screen_y.into())
+    fn screen_coordinates(&self) -> ScreenPoint {
+        self.inner.screen_coordinates()
     }
 
-    pub fn coordinates(&self) -> Coordinates {
-        Coordinates::new(
-            self.screen_coordinates(),
-            self.client_coordinates(),
-            self.element_coordinates(),
-            self.page_coordinates(),
-        )
+    fn coordinates(&self) -> Coordinates {
+        self.inner.coordinates()
     }
 
     /// The set of modifier keys which were pressed when the event occurred
-    pub fn modifiers(&self) -> Modifiers {
-        let mut modifiers = Modifiers::empty();
-
-        #[allow(deprecated)]
-        {
-            if self.alt_key {
-                modifiers.insert(Modifiers::ALT);
-            }
-            if self.ctrl_key {
-                modifiers.insert(Modifiers::CONTROL);
-            }
-            if self.meta_key {
-                modifiers.insert(Modifiers::META);
-            }
-            if self.shift_key {
-                modifiers.insert(Modifiers::SHIFT);
-            }
-        }
-
-        modifiers
+    fn modifiers(&self) -> Modifiers {
+        self.inner.modifiers()
     }
 
     /// The set of mouse buttons which were held when the event occurred.
-    pub fn held_buttons(&self) -> MouseButtonSet {
-        #[allow(deprecated)]
-        decode_mouse_button_set(self.buttons)
+    fn held_buttons(&self) -> MouseButtonSet {
+        self.inner.held_buttons()
     }
 
     /// The mouse button that triggered the event
     ///
     // todo the following is kind of bad; should we just return None when the trigger_button is unreliable (and frankly irrelevant)? i guess we would need the event_type here
     /// This is only guaranteed to indicate which button was pressed during events caused by pressing or releasing a button. As such, it is not reliable for events such as mouseenter, mouseleave, mouseover, mouseout, or mousemove. For example, a value of MouseButton::Primary may also indicate that no button was pressed.
-    pub fn trigger_button(&self) -> Option<MouseButton> {
-        #[allow(deprecated)]
-        Some(MouseButton::from_web_code(self.button))
+    fn trigger_button(&self) -> Option<MouseButton> {
+        self.inner.trigger_button()
     }
 }
 
-impl Debug for MouseData {
-    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
-        f.debug_struct("MouseData")
-            .field("coordinates", &self.coordinates())
-            .field("modifiers", &self.modifiers())
-            .field("held_buttons", &self.held_buttons())
-            .field("trigger_button", &self.trigger_button())
-            .finish()
+impl PartialEq for MouseData {
```

Comment:
Sorry I missed this initially, there isn't an underlying type for MouseData that we can compare. This could be comparing a web event to a web event. On desktop, liveview and TUI the type is SerializedPointInteraction, but on the web renderer, that struct is not included in the binary because the serde feature is turned off (to make the binary smaller).

This will be a more expensive `PartialEq` than a simple struct but not because of the derefs. That would be very cheap. Converting the underlying event to the Dioxus types in each of the functions in the trait is expensive. There is a tradeoff here between performance and binary size. If we know that we will read every field in the event, then we should convert the event once and then use the serialized event for PartialEq, Debug, etc. (this is what the 0.4 release of dioxus does). However, most of the time we only use some of the fields from the event so we are converting some data we never use. If the whole application never uses a field, we also include more code in the final binary than we need to. In this PR, I convert each field when you need the data which can help with performance and binary size if you only use a few fields, but if you read a field twice it will be slower. The middleground here is to lazily convert each field which this PR doesn't implement yet, but with the changes in this PR it should be a non-breaking change in a future PR.